packages/crawler/src/authenticator/browser-authenticator.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
This is specific implementation for the portal.azure.com. The better approach would be to have generic interface and specific implementation with corresponding class name. May be create auth provider based on auth website underlying mechanism?

Defining a function and function signature type instead of `const` will help. This code is more javascript rather than typescript.

Reply:
> This is specific implementation for the portal.azure.com. The better approach would be to have generic interface and specific implementation with corresponding class name.

Absolutely agree! However, we don't plan to support other authentication flows at this time. It seemed overengineered to add an unnecessary abstraction layer when our goal is to always use Microsoft SSO inside a Microsoft network, limited to usage with our ADO Extension.

Reply:
Hey Madalyn! Sorry, you may not have had the full context here, it was likely never discussed in any meetings with the whole team. 

We do want to add at least dSTS auth in the future (That's what is used when logging in with a PME account), and also possibly the login.live.com auth, which is outside of internal AAD and wouldn't work with the AAD SSO we have implemented here. So it's probably justifiable to create an abstraction here.